test/view/tournament/tournament_screen_test.dart

```diff
@@ -0,0 +1,180 @@
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:lichess_mobile/src/model/auth/auth_session.dart';
+import 'package:lichess_mobile/src/model/common/chess.dart';
+import 'package:lichess_mobile/src/model/common/id.dart';
+import 'package:lichess_mobile/src/model/common/perf.dart';
+import 'package:lichess_mobile/src/model/common/time_increment.dart';
+import 'package:lichess_mobile/src/model/tournament/tournament.dart';
+import 'package:lichess_mobile/src/model/tournament/tournament_repository.dart';
+import 'package:lichess_mobile/src/model/user/user.dart';
+import 'package:lichess_mobile/src/network/socket.dart';
+import 'package:lichess_mobile/src/styles/lichess_icons.dart';
+import 'package:lichess_mobile/src/view/tournament/tournament_screen.dart';
+import 'package:mocktail/mocktail.dart';
+
+import '../../network/fake_websocket_channel.dart';
+import '../../test_provider_scope.dart';
+
+const kTournamentId = TournamentId('test');
+
+const kTournament = Tournament(
+  id: kTournamentId,
+  createdBy: 'tom-anders',
+  timeIncrement: TimeIncrement(3, 0),
+  perf: Perf.blitz,
+  variant: Variant.standard,
+  berserkable: true,
+  featuredGame: null,
+  fullName: 'Test Tournament',
+  description: null,
+  isFinished: false,
+  isStarted: true,
+  timeToStart: null,
+  timeToFinish: Duration(minutes: 30),
+  standing: null,
+  me: null,
+  nbPlayers: 12,
+  verdicts: (list: IList.empty(), accepted: true),
+  reloadEndpoint: null,
+);
+
+StandingPlayer makeTestPlayer(int index) => StandingPlayer(
+  user: LightUser(id: UserId.fromUserName('player$index'), name: 'player$index'),
+  rating: 1500 + index,
+  score: index,
+  provisional: false,
+  withdraw: index.isOdd,
+  sheet: (fire: index.isEven, scores: const IList.empty()),
+);
+
+final kFirstStandingPage = (page: 1, players: Iterable.generate(10, makeTestPlayer).toIList());
+final kSecondStandingPage = (
+  page: 2,
+  players: Iterable.generate(2, (i) => makeTestPlayer(i + 10)).toIList(),
+);
+
+class MockTournamentRepository extends Mock implements TournamentRepository {}
+
+void main() {
+  setUpAll(() {
+    registerFallbackValue(kTournament);
+  });
+
+  group('Tournament screen', () {
+    final isOurTestTournament = isA<Tournament>().having((t) => t.id, 'id', kTournamentId);
+
+    testWidgets('Displays tournament data and standings', (WidgetTester tester) async {
+      final mockRepository = MockTournamentRepository();
```

Comment:
I prefer to mock the http client and the server responses instead, in order to test JSON decoding (it can help to track differences between what the app expects and what the server returns).

Reply:
Ah, yeah I found that to be a bit cumbersome, especially when I need to modify the data (e.g. change the gameId, standings, etc.)

Would you be okay if I add tests with real API responses for the `TournamentRepository` instead?

I think for studies we also did this: one test with real data for the repository, and then mock the repository in the screen test

Reply:
Right, I wasn't sure if using strong interpolation would be considered "cheating" since then you're not really using the real world server responses anymore, but alright, will keep it in mind for the next time :)